docs/writing-rules/testing-rules.md

```diff
@@ -31,6 +31,29 @@ Other than annotations there are three things to remember when creating tests:
 The `.test.yaml` file extension can also be used for test files. This is necessary when testing YAML language rules.
 :::
 
+## Testing autofix
+
+Semgrep's testing mechanism also provides a way to test the behavior of any `fix` values defined in the rules.
+To define a test for autofix behavior: 
+
+1. Create a new **autofix test file** with the `.fixed` suffix before the file type extension.  
+   For example, the autofix test file for a rule with test code in `path/to/rule.py` will be named `path/to/rule.fixed.py`.
+2. Within the autofix test file, enter the expected result of applied autofix rule to the test code.
+3. Run `semgrep --test` to verify that your autofix test file is correctly detected.
+
+When you use `semgrep --test`, Semgrep applies the autofix rule to the original test code (`path/to/rule.py`), and then verifies whether this matches the expected outcome defined in the autofix test file (`path/to/rule.fixed.py)`. If there is a mismatch, the line diffs are printed.```
+   If there is a mismatch, the line diffs will be printed.
```

Comment:
```suggestion

```
Looks like the line "If there is a mismatch..." was printed twice (see above). Suggesting to remove the second one.